plugins/modules/system/filesystem.py

```diff
@@ -16,18 +16,28 @@
 description:
   - This module creates a filesystem.
 options:
+  state:
+    description:
+    - If C(state=present), the filesystem is created if it doesn't already
+      exist, that is the default behaviour if I(state) is omitted.
+    - If C(state=absent), all filesystem signatures on I(dev) are wiped.
+    - When C(state=absent), all other options but I(dev) are ignored, and the
+      module doesn't fail if the device I(dev) doesn't actually exist.
+    choices: [ present, absent ]
```

Comment:
```suggestion
    choices: [ present, absent ]
    type: str
```

Reply:
Please note that for documentation, default is "no type", which is wrong. For documentation you must explicitly put the type everywhere. (Not doing that is a sanity error, which is probably on the ignore list for this module because it is already broken with this regard.)

Reply:
you finally convinced me :)
done everywhere (unless I missed something)